lib/AST/ASTContext.cpp

```diff
@@ -3297,10 +3297,6 @@ SILFunctionType::SILFunctionType(
   }
         
   if (genericSig) {
-    assert(!genericSig->areAllParamsConcrete() &&
-           "If all generic parameters are concrete, SILFunctionType should "
-           "not have a generic signature at all");
-
```

Comment:
Is there any way to assert that the original declaration is non-generic? It does not seem to be possible to recover it from here..